agent/ui_endpoint.go

```diff
@@ -199,7 +203,25 @@ RPC:
 		return nil, err
 	}
 
-	return summarizeServices(out.Dump, s.agent.config), nil
+	summary := summarizeServices(out.Dump, s.agent.config)
+
+	// TODO(cpiraino): Make this better!
+	// We explicitly remove the gateway service from the summary b/c
+	// summarizeServices will add a element representing the gateway into the
+	// list. This is useful when dumping out all of the services available to
+	// consul, where we construct information about how many services are
+	// associated to each gateway.
+
+	// However, since we are the UIGatewayServicesNodes endpoint does not want
+	// the same semantics, we do this here.
+	for i, s := range summary {
+		if s.Name == args.ServiceName {
+		summary = append(summary[:i], summary[i+1:]...)
+		break
+		}
+	}
```

Comment:
I really don't like that I had to do this, but I tried a bunch of different ways of refactoring `summarizeServices` and nothing really worked well. I think that we are just using `summarizeServices` for two different usecases:
 1) dump all the services that Consul knows about and summarize
 2) dump all the services associated to a gateway and summarize

And this PR makes those two usecases diverge from each other.